### PR TITLE
Remove null values.

### PR DESCRIPTION
**Pull request by Natim**
This is a first step in order to build a Kinto compatible JSONSchema.

### CODE REVIEW

File: jobs/buildhub/scrape_archives.py [hunk 1]
```diff
@@ -112,9 +89,25 @@ def archive(product, version, platform, locale, url, size, date, metadata=None):
...
+        channel = metadata.get("moz_update_channel", channel)
+        repository = metadata["moz_source_repo"].replace("MOZ_SOURCE_REPO=", "")
+        record['source']['repository'] = repository
+        record['source']['tree'] = repository.split("hg.mozilla.org/", 1)[-1]
+        buildid = metadata["buildid"]
+        builddate = datetime.datetime.strptime(buildid[:12], "%Y%m%d%H%M").isoformat()
+        record['build'] = {
+            "id": buildid,
+            "date": builddate,
+        }
```
Comment (leplatrem) [inline jobs/buildhub/scrape_archives.py]: This is a micronit, but could you group together the obtention of metadata values and appart the assignations into `record`?

File: ui/Decoder.elm [hunk 2]
```diff
@@ -28,26 +28,27 @@ buildDecoder =
...
 
 sourceDecoder : Decoder Source
 sourceDecoder =
     decode Source
         |> required "product" string
-        |> required "tree" string
-        |> required "revision" (nullable string)
+        |> optional "tree" (nullable string) Nothing
+        |> optional "revision" (nullable string) Nothing
+        |> optional "repository" (nullable string) Nothing
```
Comment (leplatrem) [inline ui/Decoder.elm]: Shouldn't be `string Nothing` only? Remove `nullable`?

**Comment by Natim** [comment]
> Did leave nullable in Elm types for a particular reason?

Yes this was to be able to decode to a Maybe and use ``Nothing`` as the default value in the decoder.
So that if the field is not present we default its value to null in the Elm type.

File: ui/Decoder.elm (hunk 2 above)
Comment (magopian) [inline ui/Decoder.elm]: The elm-decode-pipeline needs the decoder to be of the same type as the fallback, so a `Maybe` in both case. And the only way (that I know of) to do that is to use `nullable`.

File: ui/Model.elm [hunk 3]
```diff
@@ -118,11 +118,11 @@ applyFilters model =
         |> List.filter
             (\buildRecord ->
                 (recordStringEquals (.source >> .product) model.productFilter) buildRecord
-                    && (recordMaybeStringEquals (.target >> .version) model.versionFilter) buildRecord
+                    && (recordStringEquals (.target >> .version) model.versionFilter) buildRecord
                     && (recordStringEquals (.target >> .platform) model.platformFilter) buildRecord
-                    && (recordMaybeStringEquals (.target >> .channel) model.channelFilter) buildRecord
```
Comment (magopian) [inline ui/Model.elm]: I believe `recordMaybeStringEquals` isn't used anywhere else, so you can remove its code entirely.

File: ui/View.elm [hunk 4]
```diff
@@ -314,7 +325,7 @@ viewSourceDetails source =
                 Just revision ->
                     let
                         mozillaHgUrl =
-                            treeToMozillaHgUrl source.tree
+                            treeToMozillaHgUrl <| Maybe.withDefault "" source.tree
```
Comment (magopian) [inline ui/View.elm]: Would it make sense to change the signature of `treeToMozillaHgUrl` to take a `Maybe String` instead?

File: ui/View.elm [hunk 5]
```diff
@@ -337,39 +348,35 @@ viewSourceDetails source =
             , tbody []
                 [ tr []
                     [ td [] [ text source.product ]
-                    , td [] [ text source.tree ]
+                    , td [] [ text <| Maybe.withDefault "" source.tree ]
```
Comment (magopian) [inline ui/View.elm]: for the `record.build` above you use "unknown" as the default, do you want to use the same here?

**Comment by magopian** [comment]
Thanks for this @Natim <3

**Pull request merged**